### DISCUSSION TITLE
How to get results directly in brightway2 using the calculation setup already made in activity browser

### DISCUSSION THREAD

**Discussion by zeeshankhan2** [LCA-ActivityBrowser/activity-browser#1032] [General]
Hi There,
I built a calculation setup in activity browser having multiple activities and methods and now want to get results in brightway2 for further analysis. I found the MultiLCA method to do this as below.

mlca = MultiLCA("Aluminium and Steel")          #where as "Aluminium and steel" is the name of calculation setup built in Ab
mlca.results

but when i run above code i am getting the following error. Could someone help me with the code i am new to python.

---------------------------------------------------------------------------
```python
PyPardisoError                            Traceback (most recent call last)
[d:\intro](file:///D:/intro) to brightway2 activities.ipynb Cell 10 in ()
----> [1](vscode-notebook-cell:/d%3A/intro%20to%20brightway2%20activities.ipynb#X12sZmlsZQ%3D%3D?line=0) mlca = MultiLCA("Aluminium and Steel")
      [2](vscode-notebook-cell:/d%3A/intro%20to%20brightway2%20activities.ipynb#X12sZmlsZQ%3D%3D?line=1) mlca.results

File [c:\Users\M.](file:///C:/Users/M.) Zeeshan\anaconda3\lib\site-packages\bw2calc\multi_lca.py:42, in MultiLCA.__init__(self, cs_name, log_config)
     36 self.lca = LCA(demand=self.all, method=self.methods[0], log_config=log_config)
     37 self.lca.logger.info({
     38     'message': 'Started MultiLCA calculation',
     39     'methods': list(self.methods),
     40     'functional units': [wrap_functional_unit(o) for o in self.func_units]
     41 })
---> 42 self.lca.lci(factorize=True)
     43 self.method_matrices = []
     44 self.results = np.zeros((len(self.func_units), len(self.methods)))

File [c:\Users\M.](file:///C:/Users/M.) Zeeshan\anaconda3\lib\site-packages\bw2calc\lca.py:342, in LCA.lci(self, factorize, builder)
    340 if factorize:
    341     self.decompose_technosphere()
--> 342 self.lci_calculation()

File [c:\Users\M.](file:///C:/Users/M.) Zeeshan\anaconda3\lib\site-packages\bw2calc\lca.py:350, in LCA.lci_calculation(self)
    344 def lci_calculation(self):
    345     """The actual LCI calculation.
...
--> 286     raise PyPardisoError(pardiso_error.value)
    287 else:
    288     return np.ascontiguousarray(x)

PyPardisoError: The Pardiso solver failed with error code -3. See Pardiso documentation for details.
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?e81e7728-f701-41bf-8e83-d8b890907da7) or open in a [text editor](command:workbench.action.openLargeOutput?e81e7728-f701-41bf-8e83-d8b890907da7). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...
```

**Comment by marc-vdm** [comment]
I'm not sure from the code you shared, but it seems to me you have to first `mlca.calculate()` before you can `mlca.results()`. However, questions on how to use Brightway are best directed at the Brightway mailing lists, [e.g. this one](https://brightway.groups.io/g/beginners/messages), which will be better for helping you.

**Reply by marc-vdm** [reply]
@zeeshankhan2 Oh and perhaps also check out: https://docs.brightway.dev/en/latest/api/bw2calc/multi_lca/index.html

**Reply by zeeshankhan2** [reply]
Thanks @marc-vdm i will post it in brightway2mailing list.unfortunately I dont not see any function for mlca.calculate() may be i am missing something
Thanks

**Reply by marc-vdm** [reply]
You're right, I was mistaken there, I think there's another problem with some library versions,I see @Zoophobus can help you further with that.

**Reply by zeeshankhan2** [reply]
Thanks @marc-vdm

**Comment by Zoophobus** [comment]
HI @zeeshankhan2 please ignore this!! could you provide a printout of your environment? The packages that we want to look out for in case it is a result of an issue with the dependencies are intel-openmp and the mkl, there's a conflict between the versions (2022.0.1 and 2021.4.0)

**Reply by zeeshankhan2** [reply]
Hi @Zoophobus .
the caclulation setup is as follow and when i run it automoatically as an input to MultiLCA it gives the above error.
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/09e53995-61bd-415d-9165-f12bb339be14)

**Reply by zeeshankhan2** [reply]
@Zoophobus do you mean this by print out of environment?

**Reply by Zoophobus** [reply]
Sorry @zeeshankhan2 I meant your conda environment

**Reply by Zoophobus** [reply]
if you do `conda list` in the same environment you are running brightway

**Reply by zeeshankhan2** [reply]
I get this after running conda list in the same environment.Sorry for such a long list 

![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/8ba882fd-9873-4a36-b644-c34064221dca)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/ffc4c785-2020-4606-984f-fd29380d86a7)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/adf4c194-e758-48a3-881c-c9a9638fa058)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/448e4c10-e71a-4ef5-ad2c-17ff4e3e0f93)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/276ef590-1866-414a-90f0-84e0bc8048a6)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/5b0a5419-db80-49c7-b7da-ba6ddd757cf9)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/7d776a52-23e1-426e-8d86-0d838e8df2c8)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/77edd4e2-ebfa-4c47-873c-2f727916c8d0)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/9e6d187f-2cd2-4bb3-84bb-6c562005b4e5)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/1a3cae69-af74-4a98-9937-892b414db1b0)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/c06de7f4-b7b9-4ed7-8d21-29122a484d87)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/d809e11e-1b6d-40d4-978e-f826d870de6e)
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/bbc198d9-dbbf-49ad-9585-71a5ec3f9680)

**Reply by Zoophobus** [reply]
so your version of intel-openmp is much more recent, so it probably isn't the same issue as I found from pypardiso, could you just check by running this code snippet in the same environment

<code>python -c "import pypardiso as pp; import numpy as np; import scipy.sparse as sp; A = sp.rand(5000, 5000, density=0.05, format='csr'); b = np.random.rand(5000); x = pp.spsolve(A, b); print(x.sum())"
</code>

**Reply by zeeshankhan2** [reply]
I get this now.
![image](https://github.com/LCA-ActivityBrowser/activity-browser/assets/86916007/c16fd142-3b5d-48af-bdda-46e6b3a33c86)

**Reply by Zoophobus** [reply]
Okay, so that may be related to this older [issue](https://github.com/haasad/PyPardisoProject/issues/36) then. You can try downgrading your version of intel-openmp with the following:

<code>conda install -c conda-forge intel-openmp=2021.4</code>

and see if that works.

I'm linking @haasad as he has more experience with this and might be interested

**Reply by zeeshankhan2** [reply]
thanks @Zoophobus .Anyway, I did the same but still same error is coming up.

**Reply by Zoophobus** [reply]
@zeeshankhan2 maybe you can post your results here to the brightway mailing list, I hope they should have better ideas than myself

**Reply by zeeshankhan2** [reply]
@Zoophobus  thanks for the input i will do the same.

**Reply by zeeshankhan2** [reply]
Hi @Zoophobus thanks for the last input i think it worked now i dont know why it was not working yesterday but now i am getting an array of results thanks.
Sharing it below for the future may be help out someone else.

#Geeting results in brightway from the calculation setup made in Ab
mlca = MultiLCA("Aluminium and Steel")   #Aluminium and Steel is name of Calculation setup
mlca.results

Output:
array([[2.09657573e+00, 2.65885607e+00, 2.32025668e+00, 2.83776633e+00,
        2.06668225e+00, 2.11006377e+00, 2.23153339e+00, 2.61858445e+00,
        2.08202261e+00, 3.88123061e-04, 8.58825500e-04, 2.22964321e-03,
        6.58157617e-03, 5.94571506e-04, 4.42790488e-04, 2.58206700e-03,
        7.66388934e-03, 2.06501616e+00, 2.10792596e+00, 2.22802181e+00,
        2.61071156e+00, 2.08014962e+00, 2.10820635e+00, 2.31764391e+00,
        2.81415585e+00, 2.10993244e+00, 2.32151897e+00, 2.82313615e+00,
        1.27796885e-03, 1.27898318e-03, 1.28193721e-03, 1.29131535e-03,
        1.27841373e-03, 1.28329727e-03, 1.29299360e-03, 1.31641269e-03,
        9.87751202e-03, 2.31825221e+00, 2.80769885e-03, 2.31415706e+00,
        1.28745586e-03, 8.36607011e+01, 7.50058280e+00, 7.56299839e+01,
        5.30134347e-01, 2.23006127e+01, 1.01162435e-03, 2.38480012e-03,
        2.27492459e-02, 1.68631197e-08, 0.00000000e+00, 9.89884526e-09,
        6.96427444e-09, 9.17407084e-08, 4.24156340e-08, 4.84256931e-08,
        1.24713521e-09, 6.86944888e-02, 8.04935242e+00, 1.66779663e-05,
        3.73702387e-08, 2.21137063e-07, 1.01512929e-02, 6.52158886e-01,
        9.87751202e-03, 2.23153339e+00, 2.22964321e-03, 2.22802181e+00,
        1.28193721e-03, 1.08835911e+01, 1.03461573e+01, 5.37433843e-01,
        2.23006127e+01, 1.01162435e-03, 2.38480012e-03, 2.27492459e-02,
        1.68631197e-08, 9.89884526e-09, 6.96427444e-09, 4.97349375e-08,
        4.84877557e-08, 1.24718179e-09, 6.86944888e-02, 8.04935242e+00,
        1.66779663e-05, 3.73702387e-08, 2.21137063e-07, 1.01512929e-02,
        6.52158886e-01, 6.47947272e-03, 2.68410609e+00, 1.88775006e-01,
        6.81224277e-02, 5.67731524e+00, 4.93436019e-01, 1.01170349e-03,
        1.34962508e-04, 1.92573441e-03, 6.57203036e-02, 5.75098852e-02,
...
        8.41145497e-02, 6.46796875e-02, 1.30594157e+01, 6.01425155e-01,
        2.54002161e+03, 1.74449456e+01, 3.03580125e+00, 4.49274782e-03,
        3.62829901e-04, 2.09379666e+02, 1.75757082e+03, 3.35176669e-01,
        1.65427819e-01, 2.24477814e-01, 4.50637338e-06, 2.85324962e-02,
        3.77558344e-02, 3.83420175e-02, 8.41145497e-02]])
